migrations/sql/V2019.12.21.23.39__create_mine_incident_category_codes_table.sql

```diff
@@ -0,0 +1,25 @@
+ALTER TABLE mine_incident DROP CONSTRAINT IF EXISTS mine_incident_mine_incident_category_code_fkey;
+ALTER TABLE mine_incident DROP COLUMN IF EXISTS mine_incident_category_code;
+
+ALTER TABLE mine_incident_category ALTER COLUMN mine_incident_category_code TYPE varchar(3);
+
+DROP TABLE IF EXISTS mine_incident_category_xref;
+CREATE TABLE mine_incident_category_xref (
+    mine_incident_id integer NOT NULL,
+    mine_incident_category_code varchar(3) NOT NULL,
+    PRIMARY KEY(mine_incident_id, mine_incident_category_code)
+);
+
+ALTER TABLE mine_incident_category_xref OWNER TO mds;
+
+COMMENT ON TABLE mine_incident_category_xref IS 'Contains the references to the one or many mine incident category codes that have been attached to a mine incident record.';
+
+ALTER TABLE mine_incident_category_xref
+    DROP CONSTRAINT IF EXISTS mine_incident_category_xref_mine_incident_fkey CASCADE;
+ALTER TABLE mine_incident_category_xref
+    ADD CONSTRAINT mine_incident_category_xref_mine_incident_fkey FOREIGN KEY (mine_incident_id) REFERENCES mine_incident(mine_incident_id) ON UPDATE CASCADE ON DELETE CASCADE;
+
+ALTER TABLE mine_incident_category_xref
+    DROP CONSTRAINT IF EXISTS mine_incident_category_xref_mine_incident_category_fkey CASCADE;
+ALTER TABLE mine_incident_category_xref
+    ADD CONSTRAINT mine_incident_category_xref_mine_incident_category_fkey FOREIGN KEY (mine_incident_category_code) REFERENCES mine_incident_category(mine_incident_category_code) ON UPDATE CASCADE ON DELETE CASCADE;
```

Comment:
This looks great and should run no issue. FYI Postgres has a short hand for this. 

from V2019.04.08.09.32__create_incident_reports.sql.

```
CREATE TABLE table_name( 
<column>
<column>
....
    CONSTRAINT mine_guid_fkey FOREIGN KEY (mine_guid) REFERENCES mine(mine_guid) DEFERRABLE INITIALLY DEFERRED,
    CONSTRAINT mine_incident_followup_fkey FOREIGN KEY (followup_type_code) REFERENCES mine_incident_followup_type(mine_incident_followup_type_code) DEFERRABLE INITIALLY DEFERRED
);```

Reply:
Thanks. I thought about doing it shorthand too but I was just following how I saw it done in another file. The DDL that DBeaver generates would be good too.